File: vidaSegura/src/infrastructure/auth-service.js
/**
 * AuthService - Servicio de autenticación con Form.io
 * Infraestructura - Lógica de autenticación
 */

import { httpClient } from './http-client.js';
import { FORMIO_CONFIG, ENDPOINTS, HEADERS } from '../shared/config.js';
import { logger } from '../shared/logger.js';

export class AuthService {
    constructor() {
        this.baseUrl = FORMIO_CONFIG.BASE_URL;
        this.token = null;
    }

    /**
     * Autentica usuario con email y contraseña
     */
    async loginWithCredentials(email, password) {
        try {
            logger.info('🔐 Authenticating user...');

            const url = `${this.baseUrl}${ENDPOINTS.LOGIN}`;
            const response = await httpClient.post(
                url,
                { email, password },
                HEADERS.JSON
            );

            this.token = response.token || response._id;
            logger.success('Authentication successful!', {
                token: this.token?.substring(0, 50) + '...'
            });

            return this.token;
        } catch (error) {
            logger.error('Login failed', error);
            throw error;
        }
    }

    /**
     * Obtiene el token actual
     */
    getToken() {
        return this.token;
    }

    /**
     * Verifica si hay token activo
     */
    isAuthenticated() {
        return !!this.token;
    }

    /**
     * Limpia el token (logout)
     */
    logout() {
        this.token = null;
        logger.success('User logged out');
    }
}

export const authService = new AuthService();

File: vida-segura-formio-python/src/infrastructure/auth_service.py
"""
AuthService - Servicio de autenticación con Form.io
Infraestructura - Lógica de autenticación
"""

from src.infrastructure.http_client import http_client
from src.shared.config import FORMIO_CONFIG, ENDPOINTS, HEADERS
from src.shared.logger import logger


class AuthService:
    """Servicio de autenticación para Form.io"""
    
    def __init__(self):
        """Inicializa el servicio de autenticación"""
        self.base_url = FORMIO_CONFIG.BASE_URL
        self.token = None
    
    def login_with_credentials(self, email, password):
        """
        Autentica usuario con email y contraseña
        
        Args:
            email (str): Email del usuario
            password (str): Contraseña del usuario
        
        Returns:
            str: Token de autenticación
        
        Raises:
            Exception: Si la autenticación falla
        """
        try:
            logger.info('🔐 Authenticating user...')
            
            url = f"{self.base_url}{ENDPOINTS.LOGIN}"
            response = http_client.post(
                url,
                {'email': email, 'password': password},
                HEADERS.JSON
            )
            
            self.token = response.get('token') or response.get('_id')
            logger.success('Authentication successful!', {
                'token': self.token[:50] + '...' if self.token else 'None'
            })
            
            return self.token
        
        except Exception as error:
            logger.error('Login failed', error)
            raise
    
    def get_token(self):
        """
        Obtiene el token actual
        
        Returns:
            str: Token de autenticación o None
        """
        return self.token
    
    def is_authenticated(self):
        """
        Verifica si hay token activo
        
        Returns:
            bool: True si está autenticado, False en caso contrario
        """
        return bool(self.token)
    
    def logout(self):
        """Limpia el token (logout)"""
        self.token = None
        logger.success('User logged out')


# Instancia singleton
auth_service = AuthService()
